### PR TITLE
Dịch "weight-decay_vn" - Phần 5

### LINKED ISSUE #929: Dịch "weight-decay_vn" - Phần 5
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_multilayer-perceptrons/weight-decay_vn.md#L310-L408

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by azraelzhor**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #929

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 1]
```diff
@@ -361,7 +364,7 @@ train(lambd=3)
 ## Concise Implementation
 -->
 
-## *dịch tiêu đề phía trên*
+## Cách lập trình chính xác
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
## Cách lập trình súc tích
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 2]
```diff
@@ -370,7 +373,10 @@ Moreover, this integration serves a computational benefit, allowing implementati
 Since the weight decay portion of the update depends only on the current value of each parameter, and the optimizer must to touch each parameter once anyway.
 -->
 
-*dịch đoạn phía trên*
+Bởi vì phân rã trọng số có ở khắp mọi nơi trong tối ưu mạng nơ-ron,
+Gluon làm nó trở nên thuận tiện, bằng cách tích hợp phân rã trọng số vào chính giải thuật tối ưu để dễ sử dụng khi kết hợp với bất kì hàm mất mát nào.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Gluon làm cho việc áp dụng kĩ thuật này rất thuận tiện, bằng cách tích hợp phân rã trọng số vào chính giải thuật tối ưu để có thể kết hợp với bất kì hàm mất mát nào.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 3]
```diff
@@ -370,7 +373,10 @@ Moreover, this integration serves a computational benefit, allowing implementati
...
+Hơn nữa, sự tích hợp này cũng đem lại lợi ích tính toán, cho phép các thủ thuật lập trình thêm phân rã trọng số vào giải thuật, mà không làm tăng tổng chi phí tính toán.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Hơn nữa, việc tích hợp này cũng đem lại lợi ích tính toán, cho phép ta sử dụng các thủ thuật lập trình để thêm phân rã trọng số vào thuật toán mà không làm tăng tổng chi phí tính toán.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 4]
```diff
@@ -370,7 +373,10 @@ Moreover, this integration serves a computational benefit, allowing implementati
...
+Nguyên nhân là do phần phân rã trọng số của một cập nhật chỉ phụ thuộc vào giá trị hiện tại của mỗi tham số, và dù sao thì bộ tối ưu hoá cũng phải tiếp xúc với mỗi tham số một lần.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Nguyên nhân là do tại mỗi bước cập nhật, phần phân rã trọng số chỉ phụ thuộc vào giá trị hiện tại của mỗi tham số và bộ tối ưu hoá đằng nào thì cũng phải đụng tới mỗi tham số một lần.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 5]
```diff
@@ -379,7 +385,10 @@ Note that the hyperparameter `wd` will be multiplied by `wd_mult` when updating
 Thus, if we set `wd_mult` to $0$, the bias parameter $b$ will not decay.
 -->
 
-*dịch đoạn phía trên*
+Trong đoạn mã nguồn sau đây, chúng ta sẽ xác định trực tiếp các siêu tham số phân rã trọng số thông qua `wd` khi khởi tạo `Trainer` của chúng ta.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Trong đoạn mã nguồn sau đây, chúng ta chỉ định trực tiếp siêu tham số phân rã trọng số thông qua giá trị `wd` khi khởi tạo `Trainer`.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md (hunk 1 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/weight-decay_vn.md]: okay

File: chapter_multilayer-perceptrons/weight-decay_vn.md (hunk 2 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/weight-decay_vn.md]: "rất" nghe hơi cứng?

File: chapter_multilayer-perceptrons/weight-decay_vn.md (hunk 3 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/weight-decay_vn.md]: okay

File: chapter_multilayer-perceptrons/weight-decay_vn.md (hunk 4 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/weight-decay_vn.md]: okay

File: chapter_multilayer-perceptrons/weight-decay_vn.md (hunk 5 above)
Comment (azraelzhor) [reply chapter_multilayer-perceptrons/weight-decay_vn.md]: okay

File: chapter_multilayer-perceptrons/weight-decay_vn.md (hunk 2 above)
Comment (lkhphuc) [reply chapter_multilayer-perceptrons/weight-decay_vn.md]: @azraelzhor mình thay `rất` bằng `trở nên` rồi, bạn thấy sao?

Comment (azraelzhor) [reply chapter_multilayer-perceptrons/weight-decay_vn.md]: okay @lkhphuc

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 6]
```diff
@@ -342,7 +342,7 @@ train(lambd=0)
 ### Using Weight Decay
 -->
 
-### *dịch tiêu đề phía trên*
+### Sử dụng Phân rã Trọng số
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]: Từ này đã được cập nhật tại Bảng thuật ngữ.
```suggestion
### Sử dụng Suy giảm Trọng số
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 7]
```diff
@@ -351,7 +351,10 @@ This is precisely the effect we expect from regularization.
 As an exercise, you might want to check that the $\ell_2$ norm of the weights $\mathbf{w}$ has actually decreased.
 -->
 
-*dịch đoạn phía trên*
+Bên dưới, chúng ta chạy với phân rã trọng số lớn.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Bên dưới, chúng ta chạy với suy giảm trọng số lớn.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 8]
```diff
@@ -370,7 +373,10 @@ Moreover, this integration serves a computational benefit, allowing implementati
 Since the weight decay portion of the update depends only on the current value of each parameter, and the optimizer must to touch each parameter once anyway.
 -->
 
-*dịch đoạn phía trên*
+Bởi vì phân rã trọng số có ở khắp mọi nơi trong tối ưu mạng nơ-ron,
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Bởi vì suy giảm trọng số có ở khắp mọi nơi trong tối ưu mạng nơ-ron,
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 9]
```diff
@@ -370,7 +373,10 @@ Moreover, this integration serves a computational benefit, allowing implementati
...
+Gluon làm cho việc áp dụng kĩ thuật này trở nên thuận tiện, bằng cách tích hợp phân rã trọng số vào chính giải thuật tối ưu để có thể kết hợp với bất kì hàm mất mát nào.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Gluon làm cho việc áp dụng kĩ thuật này trở nên thuận tiện, bằng cách tích hợp suy giảm trọng số vào chính giải thuật tối ưu để có thể kết hợp với bất kì hàm mất mát nào.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 10]
```diff
@@ -370,7 +373,10 @@ Moreover, this integration serves a computational benefit, allowing implementati
...
+Hơn nữa, việc tích hợp này cũng đem lại lợi ích tính toán, cho phép ta sử dụng các thủ thuật lập trình để thêm phân rã trọng số vào thuật toán mà không làm tăng tổng chi phí tính toán.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Hơn nữa, việc tích hợp này cũng đem lại lợi ích tính toán, cho phép ta sử dụng các thủ thuật lập trình để thêm suy giảm trọng số vào thuật toán mà không làm tăng tổng chi phí tính toán.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 11]
```diff
@@ -370,7 +373,10 @@ Moreover, this integration serves a computational benefit, allowing implementati
...
+Nguyên nhân là do tại mỗi bước cập nhật, phần phân rã trọng số chỉ phụ thuộc vào giá trị hiện tại của mỗi tham số và bộ tối ưu hoá đằng nào thì cũng phải đụng tới mỗi tham số một lần.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Nguyên nhân là do tại mỗi bước cập nhật, phần suy giảm trọng số chỉ phụ thuộc vào giá trị hiện tại của mỗi tham số và bộ tối ưu hoá đằng nào thì cũng phải đụng tới mỗi tham số một lần.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 12]
```diff
@@ -379,7 +385,10 @@ Note that the hyperparameter `wd` will be multiplied by `wd_mult` when updating
 Thus, if we set `wd_mult` to $0$, the bias parameter $b$ will not decay.
 -->
 
-*dịch đoạn phía trên*
+Trong đoạn mã nguồn sau đây, chúng ta chỉ định trực tiếp siêu tham số phân rã trọng số thông qua giá trị `wd` khi khởi tạo `Trainer`.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Trong đoạn mã nguồn sau đây, chúng ta chỉ định trực tiếp siêu tham số suy giảm trọng số thông qua giá trị `wd` khi khởi tạo `Trainer`.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 13]
```diff
@@ -379,7 +385,10 @@ Note that the hyperparameter `wd` will be multiplied by `wd_mult` when updating
...
+Theo mặc định, Gluon phân rã đồng thời cả trọng số và hệ số điều chỉnh.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Theo mặc định, Gluon suy giảm đồng thời cả trọng số và hệ số điều chỉnh.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 14]
```diff
@@ -379,7 +385,10 @@ Note that the hyperparameter `wd` will be multiplied by `wd_mult` when updating
...
+Cần chú ý rằng siêu tham số `wd` sẽ được nhân lên với `wd_mult` khi cập nhật các tham số mô hình.
+Như vậy, nếu chúng ta thiết lập `wd_mult` bằng $0$, tham số hệ số điều chỉnh $b$ sẽ không phân rã.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Như vậy, nếu chúng ta thiết lập `wd_mult` bằng $0$, tham số hệ số điều chỉnh $b$ sẽ không suy giảm.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 15]
```diff
@@ -412,7 +421,8 @@ The plots look identical to those when we implemented weight decay from scratch.
 However, they run appreciably faster and are easier to implement, a benefit that will become more pronounced for large problems.
 -->
 
-*dịch đoạn phía trên*
+Các đồ thị này nhìn giống hệt với những đồ thị khi chúng ta lập trình phân rã trọng số từ đầu.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Các đồ thị này nhìn giống hệt với những đồ thị khi chúng ta lập trình suy giảm trọng số từ đầu.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 16]
```diff
@@ -513,7 +523,7 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 5 -->
-*
+* Lê Cao Thăng
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
* Lê Cao Thăng
* Lê Khắc Hồng Phúc
```

**Comment by azraelzhor** [comment]
@lkhphuc @duythanhvn xem lại chỗ này giúp mình với, mình phát hiện một bug nhỏ

Note that the training error increases but the test error decreases.
Hiện đã dịch là: Cần chú ý rằng lỗi huấn luyện tăng nhưng lỗi kiểm định giảm.

Lỗi kiểm tra mới đúng.

**Comment by duythanhvn** [comment]
Phần này sang phase 2 sẽ kiểm tra lại nghen, cảm ơn bạn.
